usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
I agree with Don, in practice, for large systems doing lots of snapshots, history grows rapidly